cyto_dl/image/transforms/crop_to_seg.py

```diff
@@ -0,0 +1,136 @@
+from typing import Optional, Sequence, Union
+
+import numpy as np
+import torch
+from monai.transforms import RandomizableTransform, Transform
+from skimage.measure import regionprops
+
+
+class CentroidCrop:
+    """Class for cropping patches around passed centroids in an image."""
+
+    def __init__(self, crop_size: Sequence[str], remove_edge: bool = True):
+        self.crop_size = crop_size
+        self.remove_edge = remove_edge
+
+    def centroid_to_slice(self, centroid):
+        # slice across channel dimension
+        slices = [slice(None, None)]
+        for i, c in enumerate(centroid):
+            start = int(c - self.crop_size[i] // 2)
+            end = int(c + self.crop_size[i] // 2)
+            slices.append(slice(start, end))
+        return tuple(slices)
+
+    def _filter_edge(self, centroids, shape, labels=None):
```

Comment:
Maybe a slight more compact version:
```python
def _filter_edge(self, centroids, shape, labels=None):
    centroids = np.array(centroids)
    crop_half_size = np.array(self.crop_size) // 2

    valid_mask = np.all(
        (centroids >= crop_half_size) & (centroids <= (np.array(shape) - crop_half_size)),
        axis=1
    )

    valid_centroids = centroids[valid_mask]
    if labels is None:
        return valid_centroids, None
    return valid_centroids, np.array(labels)[valid_mask]
```

Reply:
Nice! This prompted some other changes to simplify the code.